webknossos-datastore/app/com/scalableminds/webknossos/datastore/datavault/S3DataVault.scala

```diff
@@ -94,7 +98,11 @@ class S3DataVault(s3AccessKeyCredential: Option[S3AccessKeyCredential], uri: URI
 
 object S3DataVault {
   def create(remoteSourceDescriptor: RemoteSourceDescriptor): S3DataVault = {
-    val credential = remoteSourceDescriptor.credential.map(f => f.asInstanceOf[S3AccessKeyCredential])
+    val credential = remoteSourceDescriptor.credential.flatMap {
+      case f: S3AccessKeyCredential     => Some(f)
+      case f: LegacyDataVaultCredential => Some(f.toS3AccessKey)
+      case _                            => None
+    }
```

Comment:
This was a preexisting bug I encountered while testing this. The LegaccyDataVaultCredentials were not used properly in S3 case.